src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1179,4 +1183,30 @@ public static boolean canEntityDestroy(World world, BlockPos pos, LivingEntity e
         BlockState state = world.getBlockState(pos);
         return ForgeEventFactory.getMobGriefingEvent(world, entity) && state.canEntityDestroy(world, pos, entity) && ForgeEventFactory.onEntityDestroyBlock(entity, pos, state);
     }
+
+    private static final Map<IRegistryDelegate<Item>, Integer> VANILLA_BURNS = new HashMap<>();
+
+    /**
+     * Gets the burn time of this itemstack.
+     */
+    public static int getBurnTime(ItemStack stack)
+    {
+        if (stack.isEmpty())
+        {
+            return 0;
+        }
+        else
+        {
+            Item item = stack.getItem();
+            int ret = stack.getBurnTime();
+            return ForgeEventFactory.getItemBurnTime(stack, ret == -1 ? VANILLA_BURNS.getOrDefault(item.delegate, 0) : ret);
+        }
+    }
+
+    public static void updateBurns()
```

Comment:
Sync this, and do we need to keep a server/client list in case tags are not synced correctly?

Reply:
I'm not sure if it makes any sense to be synchronized, as it should only be executed on the main thread (and being synchronized would not prevent CME's as the map is not synchronized).

Reply:
The only read is not in a loop so there wouldnt be any CMEs EXCEPT this function that does a ton of writes.
It's just a safty thing on a non-critical section.